cypress/e2e/login.cy.js

```diff
@@ -0,0 +1,40 @@
+describe('User login functionality', () => {
+  beforeEach(() => {
+    cy.visit('/')
+    cy.wait(2000)
+  })
+
+  const openLoginForm = () => {
+    cy.get('#registerModal button.btn-close').click({ force: true })
+    cy.get('button[data-bs-target="#loginModal"]').eq(0).click({ force: true })
+  }
+
+  it('should switch to the login form when login button is clicked', () => {
+    openLoginForm()
+    cy.get('#loginForm').should('be.visible')
+  })
+
+  const loginUser = (email, password) => {
```

Comment:
Instead of making reusable variables, another way to achieve the same result is to use cypress commands. When you configured cypress the first time you should have gotten a folder called "cypress/support" in this folder theres instructions on how to make reusable cypress commands. You need the files cypress/support/commands.js and cypress/support/e2e.js to get it to work. You would then also need to configure your eslint to not throw errors on no-undef in this folder.

ex. a cypress command could look like:
Cypress.Commands.add("testLoginCredentials", (email, password) => {
  cy.get("#loginEmail").type(email);
  cy.get("#loginPassword").type(password);
  cy.get("#loginForm").submit();
  cy.wait(1000);
});

ex. eslint config:
  {
    "files": ["cypress/support/*.js"],
      "rules": {
        "no-undef": "warn"
      }
      
ex. your cypress/support/e2e.js file needs to contain this for the commands to be usable in other files:
      import "./commands";